src/crypto/CHIPCryptoPAL.cpp

```diff
@@ -648,5 +649,185 @@ CHIP_ERROR GenerateCompressedFabricId(const Crypto::P256PublicKey & root_public_
+                                          ByteSpan & attestationNonce, uint32_t & timestamp, ByteSpan & firmwareInfo,
+                                          ByteSpan & vendorReserved5, ByteSpan & vendorReserved6, ByteSpan & vendorReserved7,
+                                          ByteSpan & vendorReserved8)
+{
+    ByteSpan * element_array[] = { &certificationDeclaration, &attestationNonce, nullptr,          &firmwareInfo,
+                                   &vendorReserved5,          &vendorReserved6,  &vendorReserved7, &vendorReserved8 };
+
+    uint32_t validArgumentCount = 0;
+    uint32_t currentDecodeTagId = 0;
+    CHIP_ERROR TLVError         = CHIP_NO_ERROR;
+    CHIP_ERROR TLVUnpackError   = CHIP_NO_ERROR;
+    bool argExists[9];
+    TLV::TLVReader tlvReader;
+    TLV::TLVType containerType = TLV::kTLVType_Structure;
+
+    tlvReader.Init(attestationElements.data(), static_cast<uint32_t>(attestationElements.size()));
+    ReturnErrorOnFailure(tlvReader.Next(containerType, TLV::AnonymousTag));
+    ReturnErrorOnFailure(tlvReader.EnterContainer(containerType));
+
+    memset(argExists, 0, sizeof argExists);
+
+    for (size_t i = 0; i < sizeof(element_array) / sizeof(*element_array); ++i)
+    {
+        if (element_array[i] != nullptr)
+        {
+            *element_array[i] = ByteSpan();
+        }
+    }
+
+    while ((TLVError = tlvReader.Next()) == CHIP_NO_ERROR)
+    {
+        // Since call to aDataTlv.Next() is CHIP_NO_ERROR, the read head always points to an element.
+        // Skip this element if it is not a ContextTag, not consider it as an error if other values are valid.
+        if (!TLV::IsContextTag(tlvReader.GetTag()))
+        {
+            continue;
+        }
+        currentDecodeTagId = TLV::TagNumFromTag(tlvReader.GetTag());
+        if (currentDecodeTagId < 9 && currentDecodeTagId > 0)
+        {
+            if (argExists[currentDecodeTagId])
+            {
+                // Duplicate TLV tag
+                TLVUnpackError = CHIP_ERROR_IM_MALFORMED_COMMAND_DATA_ELEMENT;
+                break;
+            }
+            else
+            {
+                argExists[currentDecodeTagId] = true;
+                validArgumentCount++;
+            }
+        }
+        switch (currentDecodeTagId)
+        {
+        case 1:
+        case 2:
+        case 4:
+        case 5:
+        case 6:
+        case 7:
+        case 8: {
+            const uint8_t * data = nullptr;
+            TLVUnpackError       = tlvReader.GetDataPtr(data);
+            if (element_array[currentDecodeTagId - 1] != nullptr)
+            {
+                *element_array[currentDecodeTagId - 1] = ByteSpan(data, tlvReader.GetLength());
+            }
+        }
+        break;
+        case 3: {
+            TLVUnpackError = tlvReader.Get(timestamp);
+        }
+        default:
+            // Unsupported tag, ignore it.
```

Comment:
Only some unsupported tags are OK. They are the vendor-specific tags. This default should have an error depending on  the validity of the schema (e.g. if it's not a fully qualified tag, after all context-specific tags, then it's an error).